Makefile

```diff
@@ -387,32 +398,66 @@ __check_defined = \
     $(if $(value $1),, \
     $(error Undefined make flag: $1$(if $2, ($2))))
 
-# Flash the compiled
-flash: $(BUILD)/$(OUT_FILE)-nosd.hex
+	
+#------------------- Flash with NRFUTIL -------------------
+ifeq ($(FLASHER),nrf)
+
+  # Flash the compiled
+  flash: $(BUILD)/$(OUT_FILE)-nosd.hex
 	@echo Flashing: $(notdir $<)
 	$(NRFJPROG) --program $< --sectoranduicrerase -f nrf52 --reset
 
-# dfu using CDC interface
-dfu-flash: $(BUILD)/$(MERGED_FILE).zip
+  # dfu using CDC interface
+  dfu-flash: $(BUILD)/$(MERGED_FILE).zip
 	@:$(call check_defined, SERIAL, example: SERIAL=/dev/ttyACM0)
 	$(NRFUTIL) --verbose dfu serial --package $< -p $(SERIAL) -b 115200 --singlebank --touch 1200
 
-erase:
+  erase:
 	@echo Erasing flash
 	$(NRFJPROG) -f nrf52 --eraseall
 
-# flash SD only
-sd:
+  # flash SD only
+  sd:
 	@echo Flashing: $(SD_HEX)
 	$(NRFJPROG) --program $(SD_HEX) -f nrf52 --sectorerase --reset
 
-# flash MBR only
-mbr:
+  # flash MBR only
+  mbr:
 	@echo Flashing: $(MBR_HEX)
 	$(NRFJPROG) --program $(MBR_HEX) -f nrf52 --sectorerase --reset
 
+#------------------- Flash with PYOCD -------------------
+else ifeq ($(FLASHER),pyocd)
+
+  # Flash the compiled
+  flash: $(BUILD)/$(OUT_FILE)-nosd.hex
+	@echo Flashing: $(notdir $<)
+	$(PYOCD) flash -t $(MCU_SUB_VARIANT) $<
+
+  # dfu unsupported with pyocd
+  dfu-flash: $(BUILD)/$(MERGED_FILE).zip
+	$(error dfu-flash is not implemented with pyocd as the FLASHER)
+
+  erase:
+	@echo Erasing flash
+	$(PYOCD) erase -t $(MCU_SUB_VARIANT) --chip
+
+  # flash SD only
+  sd:
+	@echo Flashing: $(SD_HEX)
+	$(PYOCD) flash -t $(MCU_SUB_VARIANT) $(SD_HEX)
+
+  # flash MBR only
+  mbr:
+	@echo Flashing: $(MBR_HEX)
+	$(PYOCD) flash -t $(MCU_SUB_VARIANT) $(MBR_HEX)
+
+endif
+
+#------------------- Debugging -------------------
+
 gdbflash: $(BUILD)/$(MERGED_FILE).hex
-	@echo Flashing: $<
+ 	@echo Flashing: $<

```

Comment:
this rule must remain to use tab

Reply:
Heyyy, how did that space get in there! o.o